### PR TITLE
feat: Ubuntu Noble support

### PR DESCRIPTION
**Pull request by mattwillsher**
Enhancement:

* Adds support for Ubuntu Nobel/24.04.
* Fixes various issues with tests

Reason:

* New version of Ubuntu LTS recently released.

Result:

Issue Tracker Tickets (Jira or BZ if any):

### CODE REVIEW

File: tests/tests_all_options.yml [hunk 1]
```diff
@@ -93,7 +93,7 @@
           | grep -o '^\(     \|       \)[A-Z][A-Za-z0-9]*\(.\|  \)' \
           | grep -v "[A-Za-z0-9] $" | grep -v "[^A-Za-z0-9 ]$" \
           | awk '{ print $1 }' \
-          | grep -v '^$' | grep -v "^\(Match\|OpenSSH\|The\|Arguments\|Theo\)$"
+          | grep -v '^$' | grep -v "^\(Match\|OpenSSH\|The\|Arguments\|Theo\|Tatu\|Aaron\)$"
```
[the remark this answers is not in the document]
Comment (mattwillsher) [reply tests/tests_all_options.yml:96]: I don't understand what you mean by 'One more `Each`'. Can you elaborate please?

[the remark this answers is not in the document]
Comment (Jakuje) [reply tests/tests_all_options.yml:96]: The `Each` keyword is now being detected in the ubuntu-latest run so it needs to be added here. Then I think we are good got go.

**Comment by mattwillsher** [comment]
See #290

File: tests/tests_all_options.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jakuje) [reply tests/tests_all_options.yml]: Do you know you can (force-)push into the same branch to update a PR? Anyway, it seems like in the next run, there are some more differences in the service/socket units ...

**Pull request closed without merging**